### PR TITLE
Adding function to convert Pauli terms in qubit operator to circuits

### PR DESCRIPTION
**Pull request by pediejo**

### CODE REVIEW

File: src/python/zquantum/core/circuit/_gate.py [hunk 1]
```diff
@@ -500,6 +500,25 @@ def to_qiskit(self, qreg=None):
             ]
 
         # two-qubit gates
+        if self.name == 'CNOT':
+            return [qiskit.extensions.standard.CnotGate(), [qiskit_qubits[0], qiskit_qubits[1]], []]
```
Comment (simonwa7) [inline src/python/zquantum/core/circuit/_gate.py:504]: This is repeated from line 522

File: src/python/zquantum/core/circuit/_gate.py [hunk 2]
```diff
@@ -500,6 +500,25 @@ def to_qiskit(self, qreg=None):
...
+            return [qiskit.extensions.standard.RXGate(pi/2),             [qiskit_qubits[1]],                   [],
+                    qiskit.extensions.standard.RYGate(pi-params[0]/2), [qiskit_qubits[1]],                   [],
+                    qiskit.extensions.standard.CzGate(),                 [qiskit_qubits[0], qiskit_qubits[1]], [],
+                    qiskit.extensions.standard.RYGate(-(pi-params[0]/2)),[qiskit_qubits[1]],                   [],
+                    qiskit.extensions.standard.RXGate(-pi),              [qiskit_qubits[1]],                   [],
+                    qiskit.extensions.standard.CzGate(),                 [qiskit_qubits[0], qiskit_qubits[1]], [],
+                    qiskit.extensions.standard.RXGate(pi/2),             [qiskit_qubits[1]],                   [],
+                    qiskit.extensions.standard.RZGate(params[0]/2),      [qiskit_qubits[0]],                   []]
+        if self.name == 'SWAP':
+            return [qiskit.extensions.standard.SwapGate(), [qiskit_qubits[0], qiskit_qubits[1]], []]
```
Comment (simonwa7) [inline src/python/zquantum/core/circuit/_gate.py:521]: I believe these still need to be tested

File: src/python/zquantum/core/circuit/_gate.py [hunk 3]
```diff
@@ -620,6 +639,12 @@ def to_qiskit(self, qreg=None):
                 [qiskit_qubits[0], qiskit_qubits[1]],
                 [],
             ]
+        if self.name == 'CU3':
+            return [ 
+                  qiskit.extensions.standard.CU3Gate(params[0], params[1], params[2]),
+                  [qiskit_qubits[0], qiskit_qubits[1]],
+                  [],
+            ]
```
Comment (simonwa7) [inline src/python/zquantum/core/circuit/_gate.py:647]: Also needs to be tested

File: src/python/zquantum/core/utils.py [hunk 4]
```diff
@@ -387,6 +387,19 @@ def save_list(array, filename, artifact_name=""):
...
 
+def save_generic_dict(dictionary, filename):
+    """Save dictionary as json
+
+    Args:
+        dictionary (dict): the dict containing the data
+    """
+    dictionary_stored = {}
+    dictionary_stored['schema'] = SCHEMA_VERSION + '-dict'
+    dictionary_stored['dict'] = dictionary
```
Comment (simonwa7) [inline src/python/zquantum/core/utils.py:398]: This seems a bit odd - why not just put `'schema'` as one of the keys in the original dict? Pinging @mstechly for input

**Review by simonwa7 — CHANGES_REQUESTED** [review]

**Comment by pediejo** [comment]
This pull request has been superseded by https://github.com/zapatacomputing/z-quantum-core/pull/138. So I am closing it. @akataba, if those early commits of yours on this branch are no longer needed, then we should delete this branch.

**Pull request closed without merging**